crates/integration_tests/testdata/spark/provision.py

```diff
@@ -18,7 +18,13 @@
 from pyspark.sql import SparkSession
 from pyspark.sql.functions import current_date, date_add, expr
 
-spark = SparkSession.builder.getOrCreate()
+spark = (
+    SparkSession
+        .builder
+        .config("spark.sql.shuffle.partitions", "1")
```

Comment:
This is important, otherwise we get many small parquet files with a single row. When a positional delete hits the Parquet file with one row, the parquet file gets dropped instead of having a merge-on-read delete file.

Reply:
nit: add this comment alongside the code itself

also curious why its needed here, we use the same logic in pyiceberg https://github.com/apache/iceberg-python/blob/1278e8880c4767287dc69208ced20bd444c37228/dev/provision.py#L25

Reply:
I recall that I ran into the same thing with PyIceberg, but not sure how I fixed it there. Maybe good to add it there as well. Having a lot of partitions doesn't make any sense anyway, and will slow down the tests.

Reply:
yep! 2 birds 1 stone :) 

https://github.com/apache/iceberg-python/pull/1417#issuecomment-2555729306

Reply:
I forgot to `git add` the comment 🤦 It is there now